task-maker-format/src/ioi/sanity_checks/task.rs

```diff
@@ -121,3 +125,95 @@ impl SanityCheck for NoBitsStdCpp {
         Ok(())
     }
 }
+
+/// Check that "implied" subtasks dependecies do not form cycles.
+#[derive(Debug, Default)]
+pub struct SubtaskDependencies;
+make_sanity_check!(SubtaskDependencies);
+
+impl SanityCheck for SubtaskDependencies {
+    type Task = IOITask;
+
+    fn name(&self) -> &'static str {
+        "SubtaskDependencies"
+    }
+
+    fn category(&self) -> SanityCheckCategory {
+        SanityCheckCategory::Task
+    }
+
+    fn post_hook(&self, task: &Self::Task, eval: &mut EvaluationData) -> Result<(), Error> {
+        let mut table = HashSet::new();
+        for &st1 in task.subtasks.keys() {
+            for &st2 in task.subtasks.keys() {
+                if st1 != st2 {
+                    table.insert((st1, st2));
+                }
+            }
+        }
+
+        for sol in &eval.solutions {
+            let mut map = HashMap::new();
```

Comment:
Same goes here. What does this map contain? I see you put 1.0, 0.0 and NaN inside of it. Furthermore, I'm not a fan of using NaN's comparison properties in any kind of logic.